llvm/test/Transforms/ConstraintElimination/pr70510.ll

```diff
@@ -0,0 +1,49 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 3
+; RUN: opt -passes=constraint-elimination -S %s | FileCheck %s
+
+define i32 @main() {
+; CHECK-LABEL: define i32 @main() {
+; CHECK-NEXT:  entry:
+; CHECK-NEXT:    br label [[FOR_HEADER:%.*]]
+; CHECK:       for.header:
+; CHECK-NEXT:    [[IND1:%.*]] = phi i64 [ [[IND1_I:%.*]], [[FOR_LATCH:%.*]] ], [ 0, [[ENTRY:%.*]] ]
+; CHECK-NEXT:    [[IND2:%.*]] = phi i64 [ [[IND2_I:%.*]], [[FOR_LATCH]] ], [ 5, [[ENTRY]] ]
+; CHECK-NEXT:    [[REM:%.*]] = srem i64 [[IND2]], 4
+; CHECK-NEXT:    [[CMP6:%.*]] = icmp eq i64 [[IND1]], [[REM]]
+; CHECK-NEXT:    br i1 [[CMP6]], label [[EXIT1:%.*]], label [[FOR_LATCH]]
+; CHECK:       for.latch:
+; CHECK-NEXT:    [[IND2_I]] = add i64 [[IND2]], 1
+; CHECK-NEXT:    [[IND1_I]] = add i64 [[IND1]], 1
+; CHECK-NEXT:    [[COND:%.*]] = icmp eq i64 [[IND1_I]], 8
+; CHECK-NEXT:    br i1 [[COND]], label [[EXIT2:%.*]], label [[FOR_HEADER]]
+; CHECK:       exit2:
+; CHECK-NEXT:    [[CMP:%.*]] = icmp eq i64 [[REM]], 0
+; CHECK-NEXT:    [[RET:%.*]] = zext i1 [[CMP]] to i32
+; CHECK-NEXT:    ret i32 [[RET]]
+; CHECK:       exit1:
+; CHECK-NEXT:    ret i32 0
+;
+entry:
+  br label %for.header
+
+for.header:
+  %ind1 = phi i64 [ %ind1.i, %for.latch ], [ 0, %entry ]
+  %ind2 = phi i64 [ %ind2.i, %for.latch ], [ 5, %entry ]
+  %rem = srem i64 %ind2, 4
+  %cmp6 = icmp eq i64 %ind1, %rem
```

Comment:
Could you also add a variant of the test where the bound is not loop-invariant but bot another add-rec, e.g. from a load or call?